go/vt/vtgate/planbuilder/testdata/reference_cases.json

```diff
@@ -0,0 +1,644 @@
+            "Keyspace": {
+              "Name": "main",
+              "Sharded": false
+            },
+            "OperatorType": "Route",
+            "Query": "select 1 from ref_in_source",
+            "Table": "ref_in_source",
+            "Variant": "Reference"
+          }
+        ],
+        "JoinColumnIndexes": "L:0",
+        "OperatorType": "Join",
+        "TableName": "`user`_ref_in_source",
+        "Variant": "Join"
+      },
+      "Original": "select user.col from user join ref_in_source",
+      "QueryType": "SELECT"
+    },
+    "gen4-plan": {
+      "Instructions": {
+        "FieldQuery": "select `user`.col from `user`, ref_with_source as ref_in_source where 1 != 1",
+        "Keyspace": {
+          "Name": "user",
+          "Sharded": true
+        },
+        "OperatorType": "Route",
+        "Query": "select `user`.col from `user`, ref_with_source as ref_in_source",
+        "Table": "`user`, ref_with_source",
+        "Variant": "Scatter"
+      },
+      "Original": "select user.col from user join ref_in_source",
+      "QueryType": "SELECT",
+      "TablesUsed": [
+        "main.ref_in_source",
+        "user.user"
+      ]
+    }
+  },
+  {
+    "comment": "join with unqualified reference respects routing rules",
+    "query": "select user.col from user join rerouted_ref",
+    "v3-plan": {
+      "Instructions": {
+        "Inputs": [
+          {
+            "FieldQuery": "select `user`.col from `user` where 1 != 1",
+            "Keyspace": {
+              "Name": "user",
+              "Sharded": true
+            },
+            "OperatorType": "Route",
+            "Query": "select `user`.col from `user`",
+            "Table": "`user`",
+            "Variant": "Scatter"
+          },
+          {
+            "FieldQuery": "select 1 from rerouted_ref where 1 != 1",
+            "Keyspace": {
+              "Name": "main",
+              "Sharded": false
+            },
+            "OperatorType": "Route",
+            "Query": "select 1 from rerouted_ref",
+            "Table": "rerouted_ref",
+            "Variant": "Reference"
+          }
+        ],
+        "JoinColumnIndexes": "L:0",
+        "OperatorType": "Join",
+        "TableName": "`user`_rerouted_ref",
+        "Variant": "Join"
+      },
+      "Original": "select user.col from user join rerouted_ref",
+      "QueryType": "SELECT"
+    },
+    "gen4-plan": {
+      "Instructions": {
+        "FieldQuery": "select `user`.col from `user`, ref as rerouted_ref where 1 != 1",
+        "Keyspace": {
+          "Name": "user",
+          "Sharded": true
+        },
+        "OperatorType": "Route",
+        "Query": "select `user`.col from `user`, ref as rerouted_ref",
+        "Table": "`user`, ref",
+        "Variant": "Scatter"
+      },
+      "Original": "select user.col from user join rerouted_ref",
+      "QueryType": "SELECT",
+      "TablesUsed": [
+        "main.rerouted_ref",
```

Comment:
The instruction seems right, but not sure this is?

Reply:
this needs to be looked into

Reply:
Done